src/sql/parts/connection/connectionDialog/connectionController.ts

```diff
@@ -43,11 +43,39 @@ export class ConnectionController implements IConnectionComponentController {
 			onSetConnectButton: (enable: boolean) => this._callback.onSetConnectButton(enable),
 			onCreateNewServerGroup: () => this.onCreateNewServerGroup(),
 			onAdvancedProperties: () => this.handleOnAdvancedProperties(),
-			onSetAzureTimeOut: () => this.handleonSetAzureTimeOut()
+			onSetAzureTimeOut: () => this.handleonSetAzureTimeOut(),
+			onFetchDatabases: (serverName: string, authenticationType: string, userName?: string, password?: string) => this.onFetchDatabases(
+				serverName, authenticationType, userName, password).then(result => {
+				return result;
+			})
 		}, providerName);
 		this._providerName = providerName;
 	}
 
+	private onFetchDatabases(serverName: string, authenticationType: string, userName?: string, password?: string): Promise<string[]> {
+		let tempProfile = this._model;
+		tempProfile.serverName = serverName;
+		tempProfile.authenticationType = authenticationType;
+		tempProfile.userName = userName;
+		tempProfile.password = password;
+		let uri = this._connectionManagementService.getConnectionId(tempProfile);
+		return new Promise<string[]>((resolve, reject) => {
+			this._connectionManagementService.connect(tempProfile, uri).then(connResult => {
+				if (connResult && connResult.connected) {
+					this._connectionManagementService.listDatabases(uri).then(result => {
+						if (result && result.databaseNames) {
+							resolve(result.databaseNames);
+						} else {
+							reject();
```

Comment:
Not sure when you would get into this situation, but should there be an error message here? Based on the error message being included below I assume it gets shown to the user when this rejects?

Reply:
This error gets swallowed. The reason this error message isn't shown to the user because it takes a lot of time to show up the error message, and ruins the UX.